sktime/pipeline/computation_setting.py

```diff
@@ -0,0 +1,8 @@
+
+
+class ComputationSetting:
```

Comment:
quick Q: what is this for - is this later going to be a builder class or config pattern?

Reply:
This class is for communication between the pipeline and its step. The pipeline and all of its steps share the same instance, and if a step is executed, it can look up in this instance for specific information that is only available after calling a method on the pipeline. E.g., consider the following scenario:


```python
pipeline = Pipeline()

pipeline.add_step(transformer, ...)
pipeline.add_step(forecaster, ...)

pipeline.fit()

pipeline.predict() # <- After this line, it is clear that on the forecaster, predict needs to be called

pipeline.predict_interval() # <- After this line, it is clear that on the forecaster, predict_interval needs to be called
```

In this scenario, the information is available after calling a method. To share such information, I introduced this class. So it is more a config class or something like that.
Also, all keyword arguments, the pipeline received during predict/transform etc., are shared via this class with its steps.  

Probably, the naming can be improved. Perhaps, there is also a better way of sharing such information.

Reply:
The only alternatives that come in my mind are:

* The pipeline sets this information in each step explicitly. A disadvantage of this solution is that these values stay there until they are deleted again. Potentially error might arise if, somewhere, the old value is not deleted after execution or not overridden in a new execution. Furthermore, we must remind where these values are set if a new element is needed, then we have to adapt it everywhere.
* We pass all values of the computation setting explicitly when calling the `get_result`. Advantage, we get rid off the computation setting class, and the validity is automatically restricted to one pipeline execution (`predict`, `fit`, .. etc. call). Furthermore, it is more explicit and clearer from where the values originate. A disadvantage is that there might be a lot of arguments in the end. However, in total, I think this is currently my preferred solution, and I also prefer this over the current solution.

Reply:
the last solution looks better from a structural perspective to me, but it breaks the "iron rule" in the sklearn-like design that `predict` and `transform` calls should not mutate the estimator - do I understand that correctly?

Reply:
I pass these arguments now through the pipeline instead of using computation_setting.